test/e2e/configurable_route_test.go

```diff
@@ -0,0 +1,286 @@
+			return false, err
+		}
+		if len(roleList.Items) != 0 {
+			return false, nil
+		}
+		return true, nil
+	}); err != nil {
+		t.Errorf("role not deleted: %v", err)
+	}
+
+	roleBindingList := &rbacv1.RoleBindingList{}
+	if err := wait.PollImmediate(1*time.Second, timeout, func() (bool, error) {
+		if err := kclient.List(context.TODO(), roleBindingList, listOptions...); err != nil {
+			return false, err
+		}
+		if len(roleBindingList.Items) != 0 {
+			return false, nil
+		}
+		return true, nil
+	}); err != nil {
+		t.Errorf("role not deleted: %v", err)
+	}
+}
+
+func pollForValidComponentRouteRole(t *testing.T, componentRoute configv1.ComponentRouteSpec) error {
+	listOptions := []client.ListOption{
+		client.MatchingLabels{
+			componentRouteHashLabelKey: util.Hash(fmt.Sprintf("%s/%s", componentRoute.Namespace, componentRoute.Name)),
+		},
+	}
+
+	roleList := &rbacv1.RoleList{}
+	err := wait.PollImmediate(1*time.Second, timeout, func() (bool, error) {
+		if err := kclient.List(context.TODO(), roleList, listOptions...); err != nil {
+			return false, nil
+		}
+
+		if len(roleList.Items) != 1 {
+			return false, nil
+		}
+
+		role := roleList.Items[0]
+		if len(role.Rules) != 1 ||
+			len(role.Rules[0].Verbs) != 3 && role.Rules[0].Verbs[0] != "get" && role.Rules[0].Verbs[1] != "list" && role.Rules[0].Verbs[2] != "watch" ||
+			len(role.Rules[0].APIGroups) != 1 && role.Rules[0].APIGroups[0] != "" ||
+			len(role.Rules[0].Resources) != 1 && role.Rules[0].Resources[0] != "secrets" ||
+			len(role.Rules[0].ResourceNames) != 1 && role.Rules[0].ResourceNames[0] != componentRoute.ServingCertKeyPairSecret.Name {
+
+			return false, fmt.Errorf("Invalid Role generated")
+		}
+
+		return true, nil
+	})
+	return err
+}
+
+func pollForValidComponentRouteRoleBinding(t *testing.T, componentRoute configv1.ComponentRouteStatus) error {
+	listOptions := []client.ListOption{
+		client.MatchingLabels{
+			componentRouteHashLabelKey: util.Hash(fmt.Sprintf("%s/%s", componentRoute.Namespace, componentRoute.Name)),
+		},
+	}
+
+	roleBindingList := &rbacv1.RoleBindingList{}
+	err := wait.PollImmediate(1*time.Second, timeout, func() (bool, error) {
+		if err := kclient.List(context.TODO(), roleBindingList, listOptions...); err != nil {
+			return false, nil
+		}
+
+		if len(roleBindingList.Items) != 1 {
+			return false, nil
+		}
+
+		roleBinding := roleBindingList.Items[0]
+		if roleBinding.RoleRef.APIGroup != "rbac.authorization.k8s.io" ||
+			roleBinding.RoleRef.Kind != "Role" ||
+			roleBinding.RoleRef.Name != roleBinding.GetName() {
+			return false, nil
+		}
+
+		if len(roleBinding.Subjects) != len(componentRoute.ConsumingUsers) {
+			return false, nil
+		}
+
+		for _, expectedSubject := range componentRoute.ConsumingUsers {
+			found := false
+			splitExpectedSubjects := strings.Split(string(expectedSubject), ":")
+			for _, subject := range roleBinding.Subjects {
+				if subject.Namespace == splitExpectedSubjects[2] {
```

Comment:
Why check namespace and not name?  The test inputs refer to two component routes that have the same namespace, so it seems like this could lead to false positives.

Reply:
Agreed, will add a check against the name.